### DISCUSSION TITLE
Problem with MP Cost

### DISCUSSION THREAD

**Discussion by Teamtorrentscripts** [Albeoris/Memoria#494] [Q&A]
Hello, I have a problem when entering the "MP Cost" script in AbilityFeatures, when modifying the MP of a spell it does not work correctly, according to this example, the Cure magic should spend half the MP while the spell is equipped. Extension accessory, both in the menu and in battle, but I tried it as is and nothing came out. Could it be that it is not installed correctly?

I just want to create an SA to modify the cost of spells to my liking without having to resort to HW.

>AA Global+
[code=Condition] CommandMenu == BattleCommandMenu_Ability1 && MPCost > 0 && CasterAccessoryId == RegularItem_Extension [/code]
[code=MPCost] MPCost / 2 [/code]

**Comment by Tirlititi** [comment]
```txt
>AA Global+
```
instead of
```txt
AA Global+
```

**Reply by Teamtorrentscripts** [reply]
Oh, I didn't realize, now it works fine. The only thing I didn't understand is because the MP cost of the spells is only affected in combat, and not in the menu as it is outside of it. The line is supposed to also affect the menu's MP, right?

**Reply by Tirlititi** [reply]
No, it's not supposed to affect the menu's MP (juste like "Half MP") because it's cheesy to just equip / unequip the Extension for using spells in the menu. And it's tedious if that's an intended behaviour.
If you really want to change the MP cost in menu as well, you'll need to resort to ``[code=Patch]`` for now.

**Reply by Teamtorrentscripts** [reply]
Ohh, now I understand so, thank you very much for the clarification ;)

And another question, what is the line so that an ally can make an attack, while all allies have the trance bar at the maximum?

Example: The Ultima spell (ID 200) causes much more damage if all allies have their trance maxed out.

**Reply by Tirlititi** [reply]
That's a bit hacky and you'd better do that using an [external battle script code](https://github.com/Albeoris/Memoria/wiki/External-Battle-scripts), but if you want to stick with ``AbilityFeatures.txt``, you can do it like that:
```txt
[code=Condition] AbilityId == 200 && BattleFilter(15, 1, 1, BattleStatus_Trance) == 15 [/code]
[code=AttackPower] AttackPower * 2 [/code]
```

**Reply by Teamtorrentscripts** [reply]
Thanks, I'll do it like that then ;)

And another thing, I'm looking for a way to make an enemy execute an ability when you leave him at 0 HP and then he dies, how would that be possible?

I have seen that the enemy "Malilith" has this Death counter option, but I copied the same code on a different enemy with a new attack and it doesn't work :S

![image](https://github.com/Albeoris/Memoria/assets/169691119/bc6fd3c8-adc3-47fb-8ef6-e6b974f11f3f)

**Reply by Tirlititi** [reply]
For that feature, there's also a flag that needs to be activated on the enemy's stats (something like "die_event", I don't remember).

**Reply by Teamtorrentscripts** [reply]
I already found it, it was by activating the "death script" option in the enemy configuration.

Thank you so much!

**Reply by Teamtorrentscripts** [reply]
Hello again, I am trying to modify the combats in Treno's armory, my problem is that when entering combat with 4 characters, the camera of the same combat does not correctly show my allies.

Can the camera angle be changed (or zoomed out) with some HW adjustment so that everyone can be seen?

![image](https://github.com/Albeoris/Memoria/assets/169691119/ebf0d4b3-6d38-4cdb-87ef-dc4cf470f17c)

**Reply by Tirlititi** [reply]
I'm afraid it cannot, no.
If there are multiple camera intros for the battle (it can be changed in the enemy "Group"), you can try using another one. A different intro means a different idling camera.

Otherwise, the only solution I see is to base your battle on another battle. Amongst the datas that you cannot move from a battle to another, there's these camera movements. So you can use another battle and change everything (including the battle background). That's a bit tedious though.

**Reply by Teamtorrentscripts** [reply]
What a shame, I will have to completely change the combat, with a different scenario. Thanks for the reply.

Do you know how it would be possible to set up a combat, with the models of the characters that the enemy "epitaph" invokes?

I have tried choosing a random enemy and changing the model to Zidane's, but when entering battle no one appears and the game crashes... I understand that it won't be as simple as just putting it in and that's it, right?

![image](https://github.com/Albeoris/Memoria/assets/169691119/fbd3b600-7474-4a0d-892a-a31bd40ebd2c)

**Reply by Tirlititi** [reply]
It should be relatively easy though.
Maybe your enemy has attack sequences that are incompatible with the new enemy model animations. If an enemy has many different attacks with many animations, it's not obvious that your "Enemy Zidane" model has that many animations. So check the animation sequence of each of its attacks and make sure everything seems OK there.
You can also go in the "Edit Resources" part to register more animations for the enemy. I think you can use the animation IDs of the normal player characters for these.

**Reply by Teamtorrentscripts** [reply]
Well, I copied exactly the same animations, model and attacks of the Zidane that Epitaph invokes in the new fight, but nothing, it crashes the same...

For now I'll leave this, since it doesn't really require much for what I wanted to do.

Now I want to know if it is possible to change the handling of the character on a certain screen of the game, for example that in a scenario, instead of controlling Zidane, you control Garnet or Vivi.

I know it's with DefinePlayerCharacter, but I don't know where to put it
